celery/canvas.py

```diff
@@ -1179,12 +1166,22 @@ def _apply_tasks(self, tasks, producer=None, app=None, p=None,
         #   XXX chord is also a class in outer scope.
         app = app or self.app
         with app.producer_or_acquire(producer) as producer:
-            for sig, res in tasks:
+            for task_index, (current_task, next_task) in enumerate(
+                lookahead(tasks)
+            ):
+                sig, res, group_id = current_task
+                _chord = sig.options.get("chord") or chord
+                if _chord is not None and next_task is None:
+                    chord_length = task_index + 1
```

Comment:
so we are going to use chord_length instead of chord_size?

Reply:
Does chord_size appear anywhere else? I remember that it does but I'm not sure.
Let's try to keep the same terminology.

Reply:
Changed `chord_length` back to `chord_size` to keep the same terminology.